src/overlays/actors/ovl_En_Rd/z_en_rd.h

```diff
@@ -8,29 +8,75 @@ struct EnRd;
 
 typedef void (*EnRdActionFunc)(struct EnRd*, GlobalContext*);
 
+#define EN_RD_GET_FLAGS(thisx) (((thisx)->params & 0xFF00) >> 8)
+
+typedef enum GibdoLimbs {
+    /*  0 */ REDEAD_GIBDO_LIMB_NONE,
+    /*  1 */ REDEAD_GIBDO_LIMB_ROOT,
+    /*  2 */ REDEAD_GIBDO_LIMB_LEFT_LEG_ROOT,
+    /*  3 */ REDEAD_GIBDO_LIMB_LEFT_THIGH,
+    /*  4 */ REDEAD_GIBDO_LIMB_LEFT_SHIN,
+    /*  5 */ REDEAD_GIBDO_LIMB_LEFT_FOOT_ROOT,
+    /*  6 */ REDEAD_GIBDO_LIMB_LEFT_FOOT,
+    /*  7 */ REDEAD_GIBDO_LIMB_RIGHT_LEG_ROOT,
+    /*  8 */ REDEAD_GIBDO_LIMB_RIGHT_THIGH,
+    /*  9 */ REDEAD_GIBDO_LIMB_RIGHT_SHIN,
+    /* 10 */ REDEAD_GIBDO_LIMB_RIGHT_FOOT_ROOT,
+    /* 11 */ REDEAD_GIBDO_LIMB_RIGHT_FOOT,
+    /* 12 */ REDEAD_GIBDO_LIMB_UPPER_BODY_ROOT,
+    /* 13 */ REDEAD_GIBDO_LIMB_TORSO_ROOT,
+    /* 14 */ REDEAD_GIBDO_LIMB_TORSO,
+    /* 15 */ REDEAD_GIBDO_LIMB_LEFT_ARM_ROOT,
+    /* 16 */ REDEAD_GIBDO_LIMB_LEFT_UPPER_ARM,
+    /* 17 */ REDEAD_GIBDO_LIMB_LEFT_FOREARM,
+    /* 18 */ REDEAD_GIBDO_LIMB_LEFT_HAND,
+    /* 19 */ REDEAD_GIBDO_LIMB_RIGHT_ARM_ROOT,
+    /* 20 */ REDEAD_GIBDO_LIMB_RIGHT_UPPER_ARM,
+    /* 21 */ REDEAD_GIBDO_LIMB_RIGHT_FOREARM,
+    /* 22 */ REDEAD_GIBDO_LIMB_RIGHT_HAND,
+    /* 23 */ REDEAD_GIBDO_LIMB_HEAD_ROOT,
+    /* 24 */ REDEAD_GIBDO_LIMB_HEAD,
+    /* 25 */ REDEAD_GIBDO_LIMB_PELVIS,
+    /* 26 */ REDEAD_GIBDO_LIMB_MAX
+} RedeadGibdoLimbs;
+
+typedef enum {
+    /* -3 */ EN_RD_TYPE_GIBDO_RISING_OUT_OF_COFFIN = -3,
+    /* -2 */ EN_RD_TYPE_GIBDO = -2,
+    /* -1 */ EN_RD_TYPE_DOES_NOT_MOURN = -1,
+    /*  0 */ EN_RD_TYPE_DOES_NOT_MOURN_IF_WALKING = 0,
+    /*  1 */ EN_RD_TYPE_REGULAR,
+    /*  2 */ EN_RD_TYPE_CRYING,
+    /*  3 */ EN_RD_TYPE_INVISIBLE
+} EnRdType;
```

Comment:
Had to cast it like this to get it to match:
```c
0, this->dyna.actor.shape.rot.y, 0, (u8)EN_RD_TYPE_GIBDO_RISING_OUT_OF_COFFIN);
```
Not sure if it's better to do that, or to make these defines instead. But I'll push this up just to show what it looks like.

Reply:
hm so the redead actor expects the lower byte as a s8, since it sign extends it to a s16 here:
```c
    if (this->actor.params & 0x80) {
        this->actor.params |= 0xFF00;
    } else {
        this->actor.params &= 0xFF;
    }
```
I think a (u8) cast is ok

there may be something more involved making more sense like `((s16)EN_RD_TYPE_GIBDO_RISING_OUT_OF_COFFIN) & 0xFF` idk, but I'm not convinced

Reply:
I experimented with defining something like so:
```c
#define EN_RD_TYPE_COFFIN 0xFD
```
But when you try to use that in EnRd, that fails:
![image](https://user-images.githubusercontent.com/12245827/162188336-97eb4562-5240-46ec-9aba-69610b466752.png)

So I think what I have now is *probably* the best we'll get? I'll leave it up to you, though.

Reply:
Sorry, didn't refresh to see your comment. `((s16)EN_RD_TYPE_GIBDO_RISING_OUT_OF_COFFIN) & 0xFF` does actually work, but for now I think I'll leave it as a `u8` cast. I think if other `Actor_Spawns` do weird stuff like this, then maybe it makes sense to make it semi-standard, but I'll keep it simple for now.

Reply:
I don't think there's a way to get rid of the assumptions made by the code in EnRd on how negative numbers are represented, so the simplest u8 cast also making those assumptions looks good to me